### PR TITLE
Bunch of small tests and some doc fixes

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Comment by kshyatt** [comment]
Did I cause that ARM failure?

File: src/execution.jl [hunk 1]
```diff
@@ -140,13 +140,15 @@ For example:
...
+    unsafe_copyto!(ad, convert(Ptr{Cvoid}, a), 10*sizeof(Float32)))
+    bd = Mem.alloc(DeviceBuffer, 10*sizeof(Float32))
+    unsafe_copyto!(bd, convert(Ptr{Cvoid}, b), 10*sizeof(Float32)))
     c = zeros(Float32, 10)
-    cd = Mem.alloc(c)
+    cd = Mem.alloc(DeviceBuffer, 10*sizeof(Float32))
 
     cudacall(vadd, (CuPtr{Cfloat},CuPtr{Cfloat},CuPtr{Cfloat}), ad, bd, cd; threads=10)
-    Mem.download!(c, cd)
+    unsafe_copyto!(onvert(Ptr{Cvoid}, c), cd, 10*sizeof(Float32)))
```
Comment (maleadt) [inline src/execution.jl]:
```suggestion
    unsafe_copyto!(convert(Ptr{Cvoid}, c), cd, 10*sizeof(Float32)))
```

File: src/memory.jl [hunk 2]
```diff
@@ -61,7 +61,8 @@ Base.convert(::Type{CuPtr{T}}, buf::DeviceBuffer) where {T} =
     Mem.alloc(DeviceBuffer, bytesize::Integer)
 
 Allocate `bytesize` bytes of memory on the device. This memory is only accessible on the
-GPU, and requires explicit calls to `upload` and `download` for access on the CPU.
+GPU, and requires explicit calls to `unsafe_copyto!`, which wraps `cudaMemCopy`,
```
Comment (maleadt) [inline src/memory.jl]:
```suggestion
GPU, and requires explicit calls to `unsafe_copyto!`, which wraps `cuMemcpy`,
```

File: src/memory.jl [hunk 3]
```diff
@@ -147,9 +154,10 @@ const HOSTREGISTER_IOMEMORY = CUDAdrv.CU_MEMHOSTREGISTER_IOMEMORY
     Mem.register(HostBuffer, ptr::Ptr, bytesize::Integer, [flags])
 
 Page-lock the host memory pointed to by `ptr`. Subsequent transfers to and from devices will
-be faster, and can be executed asynchronously. If the `MEMHOSTREGISTER_DEVICEMAP` flag is
-specified, the buffer will also be accessible directly from the GPU. These accesses are
-direct, and go through the PCI bus.
+be faster, and can be executed asynchronously. If the `HOSTREGISTER_DEVICEMAP` flag is
+specified, the buffer will also be accessible directly from the GPU. If the
+`HOSTREGISTER_PORTABLE` flag is specified, any CUDA context can access the memory.
+These accesses are direct, and go through the PCI bus.
```
Comment (maleadt) [inline src/memory.jl:159]: HOSTREGISTER_DEVICEMAP accesses go over the PCI bus, not HOSTREGISTER_PORTABLE ones, so I'd move that sentence.

**Review by maleadt — COMMENTED** [review]
Some minor changes to docs.

**Comment by maleadt** [comment]
re. the arm failure, managed memory is weird on Jetson hardware: https://github.com/rapidsai/cudf/issues/1360. might want to detect that system, or its properties.

**Comment by kshyatt** [comment]
Alright, looks like the arm build is unhappy with the device ordinal... not quite sure why

**Pull request merged**